barman/utils.py

```diff
@@ -297,30 +297,101 @@ class BarmanEncoder(json.JSONEncoder):
     """
 
+    method_list = [
+        "_to_json",
+        "_datetime_to_str",
+        "_timedelta_to_str",
+        "_decimal_to_float",
+        "binary_to_str",
+        "version_to_str",
+    ]
+
     def default(self, obj):
+        # Go through all methods until one returns something
+        for method in self.method_list:
+            res = getattr(self, method)(obj)
+            if res is not None:
+                return res
+
+        # Let the base class default method raise the TypeError
+        return super(BarmanEncoder, self).default(obj)
+
+    @staticmethod
+    def _to_json(obj):
+        """
         # If the object implements to_json() method use it
+        :param obj:
+        :return: None|str
+        """
         if hasattr(obj, "to_json"):
             return obj.to_json()
-        # Serialise date and datetime objects using ctime() method
+
+    @staticmethod
+    def _datetime_to_str(obj):
+        """
+        Serialise date and datetime objects using ctime() method
+        :param obj:
+        :return: None|str
+        """
         if hasattr(obj, "ctime") and callable(obj.ctime):
             return obj.ctime()
-        # Serialise timedelta objects using human_readable_timedelta()
+
+    @staticmethod
+    def _timedelta_to_str(obj):
+        """
+        Serialise timedelta objects using human_readable_timedelta()
+        :param obj:
+        :return: None|str
+        """
         if isinstance(obj, datetime.timedelta):
             return human_readable_timedelta(obj)
-        # Serialise Decimal objects using their string representation
-        # WARNING: When deserialized they will be treat as float values
-        # which have a lower precision
+
+    @staticmethod
+    def _decimal_to_float(obj):
+        """
+        Serialise Decimal objects using their string representation
+        WARNING: When deserialized they will be treat as float values which have a lower precision
+        :param obj:
+        :return: None|float
+        """
         if isinstance(obj, decimal.Decimal):
             return float(obj)
-        # Binary strings must be decoded before using them in
-        # an unicode string
+
+    @staticmethod
+    def binary_to_str(obj):
+        """
+        Binary strings must be decoded before using them in an unicode string
+        :param obj:
+        :return: None|str
+        """
         if hasattr(obj, "decode") and callable(obj.decode):
             return obj.decode("utf-8", "replace")
-        # Manage (Loose|Strict)Version objects as strings.
+
+    @staticmethod
+    def version_to_str(obj):
+        """
+        Manage (Loose|Strict)Version objects as strings.
+        :param obj:
+        :return: None|str
+        """
         if isinstance(obj, Version):
             return str(obj)
-        # Let the base class default method raise the TypeError
-        return super(BarmanEncoder, self).default(obj)
+
+
+class NewBarmanEncoder(BarmanEncoder):
+    @staticmethod
```

Comment:
Could you add a comment stating that this is intended to replace `BarmanEncoder` at the next major version change (assuming we're all happy with that approach)?

Reply:
Speaking of names, "new" tends to get old quite fast and I think something like BarmanEncoderV2 or similar would be better.

Reply:
Absolutely agreed. I will add comment and rename it.